CONTRIBUTING.md

```diff
@@ -145,6 +143,23 @@ These were trialed as a possible style of Python 3 rules and should not be used.
 The guidance above should be followed for new rules.
 Additionally, if you rely on a dependency that uses `_python3`-suffixed codenames, add a new rule for it that follows the guidance above.
 
+#### pip
+
+Python packages, which are only available on pip, should use the `-pip` extension. Also the `python-`/`python3-` prefix needs to be used.
```

Comment:
Did we solve the python2 vs python3 pip issue by using the `python_version` key in the rosdistro index when installing rosdeps via pip?

Reply:
@nuclearsandwich is this a question to me or to @tfoote? Because I don't understand your question at all.

Reply:
We do invoke the appropriate pip instance based on the `ROS_PYTHON_VERSION` https://github.com/ros-infrastructure/rosdep/pull/709

Which I guess does mean that we could omit the python version from the keys since it will translate to the appropriate one for your rosdistro. For the other package managers however they won't auto switch so any hybrid rules would require both versions in the package.xml with the conditional. 

So for consistency I think it makes sense to use the prefix even with the pip only packages. Technically it will cause duplication. But we're also approaching the EOL of python2 so the duplication will be short lived and then we will just have consistent rules and the pip ones will appear side by side with the non-pip ones.

Reply:
> @nuclearsandwich is this a question to me...

Sorry for the confusion @MatthijsBurgh it was directed back at the team. I ought to have said so.


> So for consistency I think it makes sense to use the prefix even with the pip only packages. Technically it will cause duplication.

I would have opted for just keeping a python3-foobar-pip key which would look odd in a python2 context but wouldn't be wrong. However if we're okay with the duplication (or doing something ugly but clever like using YAML anchors) then that's going to be better for downstream folks.